grammar/type.lyg

```diff
@@ -21,7 +21,8 @@ Type =
 
 FnSigInputs =
   | Regular:FnSigInput+ %% ","
-  | Variadic:"..."
-  | RegularAndVariadic:{ inputs:FnSigInput+ % "," "," "..." }
+  | CVariadic:FnSigCVariadicInput
+  | RegularAndCVariadic:{ inputs:FnSigInput+ % "," "," c_variadic:FnSigCVariadicInput }
   ;
-FnSigInput = { pat:Pat ":" }? ty:Type;
+FnSigInput = attrs:OuterAttr* { pat:Pat ":" }? ty:Type;
+FnSigCVariadicInput = attrs:OuterAttr* "...";
```

Comment:
This isn't right; the actual grammar is defined by:
```rust
    fn parse_ty_bare_fn(&mut self, generic_params: Vec<GenericParam>) -> PResult<'a, TyKind> {
        let unsafety = self.parse_unsafety();
        let ext = self.parse_extern()?;
        self.expect_keyword(kw::Fn)?;
        let decl = self.parse_fn_decl(|_| false, AllowPlus::No)?;
        Ok(TyKind::BareFn(P(BareFnTy { ext, unsafety, generic_params, decl })))
    }
```
Aside from that `false`, which adjusts the precedence (bias towards named vs. not), its just the regular `FnDecl` grammar, and since we don't deal with precedence here, you can use `FnDecl` instead.

Reply:
Well, I'd rather move the full grammar here than have a separate `FnDecl`, but that makes sense.
So `fn(self)` parses, I didn't know that.

Reply:
Yeah I've made some changes recently in the large scale parser refactorings to simplify things.

Reply:
You might want to review:
- https://github.com/rust-lang/rust/pull/69366
- https://github.com/rust-lang/rust/pull/69361
- https://github.com/rust-lang/rust/pull/69194
- https://github.com/rust-lang/rust/pull/69023
- https://github.com/rust-lang/rust/pull/68788
- https://github.com/rust-lang/rust/pull/68764
- https://github.com/rust-lang/rust/pull/68728
- https://github.com/rust-lang/rust/pull/67131